### PR TITLE
Library strategy

### PR DESCRIPTION
**Pull request by Vkale1**
Tried to do this as an OR SQL statement but wasn't working. It was complaining that the other option can't be None. Maybe I'm going about it wrong?
```
    def update_runs_from_accessions(self, run_accessions=None, study_accessions=None, library_strategy=None):
        jobs = Run.objects.using(self.database)
        if run_accessions and study_accessions:
            runs = jobs.filter(study__secondary_accession__in=study_accessions, primary_accession__in=run_accessions)
        else:
            runs = jobs.filter(
                Q(primary_accession__in=run_accessions) |
                Q(study__secondary_accession__in=study_accessions)
            )
        logging.info('Matched {} run(s)'.format(len(runs)))
        runs.update(library_strategy=library_strategy)
```

### CODE REVIEW

**Comment by KateSakharova** [comment]
Maybe it is easier to add **library_strategy** to _update_annotation_jobs_from_accessions_ function? I don't see any problems with that if we'll use it only for _amplicon -> raw-reads_ and _raw-reads -> amplicon_.

I understand that you created a separate function in order to skip assemblies...

**Comment by Vkale1** [comment]
@KeteSakharova So I made a separate function because library strategy is not a feature of annotation jobs tables. Rather it is a feature of 'Run' table. If we keep this separate, in the future we can add biome or anything else we want to update maybe? Open to suggestions!

File: mgnify_backlog/mgnify_handler.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import django.db
 from django.core.exceptions import ObjectDoesNotExist
 from django.db.models import Q
+from copy import deepcopy
```
Comment (KateSakharova) [inline mgnify_backlog/mgnify_handler.py]: not in use?

**Review by KateSakharova — DISMISSED** [review]

**Comment by mberacochea** [comment]
> Tried to do this as an OR SQL statement but wasn't working. It was complaining that the other option can't be None. Maybe I'm going about it wrong?
> 
> ```
>     def update_runs_from_accessions(self, run_accessions=None, study_accessions=None, library_strategy=None):
>         jobs = Run.objects.using(self.database)
>         if run_accessions and study_accessions:
>             runs = jobs.filter(study__secondary_accession__in=study_accessions, primary_accession__in=run_accessions)
>         else:
>             runs = jobs.filter(
>                 Q(primary_accession__in=run_accessions) |
>                 Q(study__secondary_accession__in=study_accessions)
>             )
>         logging.info('Matched {} run(s)'.format(len(runs)))
>         runs.update(library_strategy=library_strategy)
> ```

The problem is that the defaul values for run_accessions and study_accessions is None. You need to check that or use an empty lists as follow. Note the case when no run_accessions or study_accessions is provided, that shoud not happen IMO.

```python
def update_runs_from_accessions(self, run_accessions=None, study_accessions=None, library_strategy=None):
    jobs = Run.objects.using(self.database)
    if run_accessions and study_accessions:
        runs = jobs.filter(
            study__secondary_accession__in=study_accessions,
            primary_accession__in=run_accessions
        )
    elif run_accessions: 
        runs = jobs.filter(primary_accession__in=run_accessions)
    elif study_accessions:
        runs = jobs.filter(study__secondary_accession__in=study_accessions)
    else:
        # DANGER you are about to modify ALL the Run objects in the DB
        return

    logging.info('Matched {} run(s)'.format(len(runs)))
    runs.update(library_strategy=library_strategy)
```

**Review by KateSakharova — APPROVED** [review]
looks good now

**Review by mberacochea — APPROVED** [review]

**Pull request merged**